### PR TITLE
Implement Local storage

### LINKED ISSUE #12: Create the Edit Card
## Value Statement 

As a user 
I want to change the given information of the clinics
so I can always keep them updated or add things. 

## Description

image

## Acceptance Critera

- create a new page
- have the Input Formular 
- added the Buttons
- styled with CSS

## Tasks

- [x] create a new page
- [x] create an Input Formular
- [x] create useHistory to get back
- [x] create button to directly edit it

### LINKED ISSUE #14: Implement Local Storage
## Value Statement 

As a user 
I want to be able to add a clinic
and have it stored in my local storage.

## Acceptance Critera

- add a clinic to the local storage 
- be able to see it on the list page  
- add it to your favourites
- be able to edit the clinic
- and delete the clinic 

## Tasks

- [] implement local storage
- [] add a clinic
- [] see it on the list page 
- [] delete the clinic
- [] edit the clinic
- []safe the clinic in favourites

### PR DESCRIPTION
**Pull request by Shari-alt**
Finally I finished my sprint for this week! 
I implemented the local storage, am able to add, edit or delete a clinic and render it on the list page. 
The single view of the clinic is not styled yet but I will do this later. 
The next step is to be able to safe a clinic as a favourite. 

I am repeating a lot of code in FormAdd.js and FormEdit.js - so I would be happy to get suggestions for this. 

Than you for your code review and have a lovely weekend ☀️

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -38,7 +39,13 @@ function App() {
...
             </Route>
             <Route path="/add">
-              <Add />
+              <FormAddClinic />
+            </Route>
+            <Route path="/singleclinic/edit/:id">
+              <FormEditClinic />
+            </Route>
+            <Route path="/singleclinic/:id">
+              <SingleClinic />
```
Comment (hugh-burgess) [inline src/App.js:48]: are the params id `ide` or `id`?

File: src/components/ClinicCard.js [hunk 2]
```diff
@@ -1,30 +1,38 @@
...
 import { ReactComponent as Save } from "../images/Save.svg";
 import { ReactComponent as SavedHeart } from "../images/SavedHeart.svg";
 import { useState } from "react";
+import { useParams } from "react-router-dom";
 
-export default function ClinicCard({ name, place }) {
-  const [isClicked, setIsClicked] = useState(false);
+export default function ClinicCard({ clinicData }) {
+  const [isSaved, setIsSaved] = useState(false);
+  let { ide } = useParams();
```
Comment (hugh-burgess) [inline src/components/ClinicCard.js:12]: from this one here

File: src/components/ClinicCard.js [hunk 3]
```diff
@@ -1,30 +1,38 @@
...
-      <div className="ClinicCardButtons">
-        <button className="Button_Edit">
-          <Edit className="Edit" />
-        </button>
-        <button className="Button_Save" onClick={handleSaveClick}>
-          {isClicked ? <SavedHeart /> : <Save className="Save" />}
-        </button>
+    <article className="ClinicCardList">
+      <div className="ClinicCard">
+        <img className="ClinicPicture" src={Download} alt="Clinicpicture" />
```
Comment (hugh-burgess) [inline src/components/ClinicCard.js]: use camelCase for the alt (`clinicPicture`)

Comment (hugh-burgess) [reply src/components/ClinicCard.js]: and the className, then change in the css as needed.

File: src/components/ClinicCard.js [hunk 4]
```diff
@@ -1,30 +1,38 @@
...
+        <h3 className="ClinicName"> {clinicData.name} </h3>
+        <p className="ClinicPlace"> {clinicData.place}</p>
+        <div className="ClinicCardButtons">
+          <Link to={`/singleclinic/edit/${ide}`}>
+            <button className="Button_Edit">
+              <Edit className="Edit" />
+            </button>
+          </Link>
+          <button className="Button_Save" onClick={onfavouriteClinic}>
+            {isSaved ? <SavedHeart /> : <Save className="Save" />}
```
Comment (hugh-burgess) [inline src/components/ClinicCard.js:32]: maybe a better class name of more specificity?

File: src/components/Form.css [hunk 5]
```diff
@@ -16,7 +16,8 @@
   padding: 10px;
   padding-bottom: 40px;
   margin-top: 20px;
-  height: 100vh;
+  margin-bottom: 50px;
+  height: 70vh;
```
Comment (hugh-burgess) [inline src/components/Form.css]: check with marwin about vh, i would use `rem` instead, as VH ignores other rules and applies to the phones sizing and not the other elements or a grid for example.

File: src/components/FormAddClinic.js [hunk 6]
```diff
@@ -158,26 +182,32 @@ export default function Add() {
...
                 />
                 <label for="Bewegung"> Bewegung </label>
               </div>
 
               <div className="Checkbox">
                 <input
                   type="checkbox"
-                  value={therapy.Körper}
-                  onChange={(e) => setTherapy({ Körper: e.target.value })}
+                  checked={therapy.Körper}
```
Comment (hugh-burgess) [inline src/components/FormAddClinic.js]: maaaaybe avoid the umlaut?

File: src/components/FormAddClinic.js [hunk 7]
```diff
@@ -218,22 +251,22 @@ export default function Add() {
...
-          <input type="checkbox" value={animals} onChange={handleAnimals} />
+          <input type="checkbox" checked={animals} onChange={handleAnimals} />
           <label for="Animals"> Haustiere </label>
         </div>
 
         <div className="Room">
-          <input type="checkbox" value={room} onChange={handleRoom} />
+          <input type="checkbox" checked={room} onChange={handleRoom} />
           <label for="Room"> Einzelzimmer </label>
         </div>
```
Comment (hugh-burgess) [inline src/components/FormAddClinic.js:271]: these could be components?

File: src/components/FormEditClinic.js [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+              </div>
+            </div>
+            <div className="CenterCol">
+              <div className="Checkbox">
+                <input
+                  type="checkbox"
+                  checked={therapy.Bewegung}
+                  onChange={(e) =>
+                    setTherapy({ ...therapy, Bewegung: e.target.checked })
+                  }
```
Comment (hugh-burgess) [inline src/components/FormEditClinic.js:178]: please make this sub components for the love of god

**Review by hugh-burgess — CHANGES_REQUESTED** [review]
LGTMGMTLT. Just a couple things and make some sub components for the bigger code! :)

File: src/components/FormAddClinic.js (hunk 7 above)
Comment (Shari-alt) [reply src/components/FormAddClinic.js:271]: Thank you! I do this in the end when everything is working :)

File: src/components/FormEditClinic.js (hunk 8 above)
Comment (Shari-alt) [reply src/components/FormEditClinic.js:178]: 😂

**Pull request merged**